server/api/attendance/attendance.integration.js

```diff
@@ -0,0 +1,122 @@
+
+    var mentorSession, studentSession;
+
+    before(() => getSessionFor('mentor').then(session => mentorSession = session));
+    before(() => getSessionFor('student').then(session => studentSession = session));
+
+    let smallgroup, mentor, student;
+
+    before(() => {
+      // Add the mentor to a small group
+      return Promise.all([
+        User.findOne({"email": "mentor@mentor.com"}),
+        SmallGroup.findOne({"name": "New Small Group"}),
+        User.findOne({"email": "student@student.com"})
+      ]).then(results => {
+          mentor = results[0];
+          smallgroup = results[1];
+          student = results[2];
+          smallgroup.students.push(mentor._id);
+          smallgroup.students.push(student._id);
+          student.smallgroup = smallgroup._id;
+          return Promise.all([smallgroup.save(), student.save()]);
+        });
+    });
+
+    it('should should have put new students in the smallgroup', () => {
+      return SmallGroup.findById(smallgroup._id).then(smallgroup => {
+        expect(smallgroup.students).to.contain(mentor._id);
+        expect(smallgroup.students).to.contain(student._id);
+      });
+    });
+
+    let dayCode;
+
+    it('should allow mentors to generate an attendance code', done => {
+      mentorSession
+      .post('/api/smallgroup/'+smallgroup._id+'/daycode').end((err, res) => {
+        expect(res.body).to.have.length(6);
+        dayCode = res.body;
+        done(err);
+      });
+    });
+
+    it('should allow students to submit small group attendance', done => {
+      studentSession
+      .post('/api/attendance/attend')
+      .send({
+        "dayCode": dayCode
+      })
+      .end((err, res) => {
+        expect(res.status).to.equal(200);
+        done(err);
+      });
+    });
+
+    it('should indicate that the student\'s attendance had been taken for the small group', done => {
+      studentSession.get('/api/attendance/present/me').end((err,res) => {
+        expect(res.body[0].present).to.be.true;
+        done(err);
+      });
+    });
+
+  });
+
+  describe("Large group attendance", () => {
+
+    before(seed);
+
+    let adminSession, studentSession;
+    let dayCode;
+
+    before(() => getSessionFor('admin').then(session => adminSession = session));
+    before(() => getSessionFor('student').then(session => studentSession = session));
+
+    it('should create a large group code', done => {
+      adminSession.post('/api/classyear/daycode')
+        .send({bonusDay: false})
+        .end((err, res) =>{
+        expect(res.status).to.equal(200);
+        console.error("WARNING: res.body should be used instead of res.text");
+        dayCode = res.text;
+        done(err);
+      });
+    });
+
+    it('should allow students to submit large group attendance', done => {
+      studentSession
+      .post('/api/attendance/attend')
+      .send({
+        "dayCode": dayCode
+      })
+      .end((err, res) => {
+        expect(res.status).to.equal(200);
```

Comment:
this is closer to a smoke test- it would be a good idea to expand on these and really dig into the test cases but I would clean up the api a bit first :8ball: